OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/BaseBuilderQueueManager.cs

```diff
@@ -156,7 +156,7 @@ bool TickQueue(IBot bot, ProductionQueue queue)
 				else
 				{
 					// Check if Building is a defense and if we should place it towards the enemy or not.
-					if (actorInfo.HasTraitInfo<AttackBaseInfo>() && world.LocalRandom.Next(100) < baseBuilder.Info.PlaceDefenseTowardsEnemyChance)
+					if (baseBuilder.Info.DefenseTypes.Contains(actorInfo.Name) && world.LocalRandom.Next(100) < baseBuilder.Info.PlaceDefenseTowardsEnemyChance)
```

Comment:
This will make existing AIs not build any defenses. Can we have an update rule that reports that manual adjustments are needed on `BaseBuilderBotModule` nodes? Ideally the rule would add the actor list automatically, like https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/UpdateRules/Rules/20210321/UnhardcodeSquadManager.cs.

Reply:
rule updated